lightning/src/ln/onchaintx.rs

```diff
@@ -0,0 +1,713 @@
+					let sighash = hash_to_message!(&sighash_parts.sighash_all(&bumped_tx.input[i], &witness_script, *revoked_amount)[..]);
+					let sig = self.secp_ctx.sign(&sighash, &key);
+					bumped_tx.input[i].witness.push(sig.serialize_der().to_vec());
+					bumped_tx.input[i].witness[0].push(SigHashType::All as u8);
+					if *is_htlc {
+						bumped_tx.input[i].witness.push(pubkey.unwrap().clone().serialize().to_vec());
+					} else {
+						bumped_tx.input[i].witness.push(vec!(1));
+					}
+					bumped_tx.input[i].witness.push(witness_script.clone().into_bytes());
+					log_trace!(self, "Going to broadcast Penalty Transaction {} claiming revoked {} output {} from {} with new feerate {}...", bumped_tx.txid(), if !is_htlc { "to_local" } else if HTLCType::scriptlen_to_htlctype(witness_script.len()) == Some(HTLCType::OfferedHTLC) { "offered" } else if HTLCType::scriptlen_to_htlctype(witness_script.len()) == Some(HTLCType::AcceptedHTLC) { "received" } else { "" }, outp.vout, outp.txid, new_feerate);
+				},
+				&InputMaterial::RemoteHTLC { ref witness_script, ref key, ref preimage, ref remote_amount, ref locktime } => {
+					if !preimage.is_some() { bumped_tx.lock_time = *locktime }; // Right now we don't aggregate time-locked transaction, if we do we should set lock_time before to avoid breaking hash computation
+					let sighash_parts = bip143::SighashComponents::new(&bumped_tx);
+					let sighash = hash_to_message!(&sighash_parts.sighash_all(&bumped_tx.input[i], &witness_script, *remote_amount)[..]);
+					let sig = self.secp_ctx.sign(&sighash, &key);
+					bumped_tx.input[i].witness.push(sig.serialize_der().to_vec());
+					bumped_tx.input[i].witness[0].push(SigHashType::All as u8);
+					if let &Some(preimage) = preimage {
+						bumped_tx.input[i].witness.push(preimage.clone().0.to_vec());
+					} else {
+						bumped_tx.input[i].witness.push(vec![0]);
+					}
+					bumped_tx.input[i].witness.push(witness_script.clone().into_bytes());
+					log_trace!(self, "Going to broadcast Claim Transaction {} claiming remote {} htlc output {} from {} with new feerate {}...", bumped_tx.txid(), if preimage.is_some() { "offered" } else { "received" }, outp.vout, outp.txid, new_feerate);
+				},
+				&InputMaterial::LocalHTLC { .. } => {
+					//TODO : Given that Local Commitment Transaction and HTLC-Timeout/HTLC-Success are counter-signed by peer, we can't
+					// RBF them. Need a Lightning specs change and package relay modification :
+					// https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-November/016518.html
+					return None;
+				}
+			}
+		}
+		log_trace!(self, "...with timer {}", new_timer);
+		assert!(predicted_weight >= bumped_tx.get_weight());
+		Some((new_timer, new_feerate, bumped_tx))
+	}
+
+	pub(super) fn block_connected<B: Deref, F: Deref>(&mut self, txn_matched: &[&Transaction], claim_requests: HashMap<Sha256dHash, Vec<ClaimRequest>>, height: u32, broadcaster: B, fee_estimator: F) -> Vec<SpendableOutputDescriptor>
```

Comment:
claim_requests seems pretty redundant as-is - its a map from previous-txid to a vec of a struct which contains the previous-txid in the outpoint. Can we drop the duplication to make it a bit easier to see correctness?

Reply:
Yeah, my initial worry was someone inadvertently introducing cross-block aggregation, segmenting by txid was the reason.

Reply:
Right, but why not just have it be a map from previous-txid to something that contains the vout, instead of a full OutPoint?